connect/proxy/proxy.go

```diff
@@ -126,6 +137,36 @@ func (p *Proxy) Serve() error {
 	}
 }
 
+// removeOldListeners removes listeners from `active` that aren't referenced in
+// newUpstreams. The removed listeners are closed and any errors are logged.
+// This should be called BEFORE adding new listeners from a new configuration so
+// that if a particular port is switched from one target service to another
+// service it will correctly be re-tasked.
+func (p *Proxy) removeOldListeners(active map[string]*Listener, newUpstreams []UpstreamConfig) {
+	oldUpstreamsToRemove := map[string]bool{}
+	// Initially assume that we should remove all of the active upstreams.
+	for active := range active {
+		oldUpstreamsToRemove[active] = true
+	}
+	// Mark all the new configurations so that we don't remove one that hasn't
```

Comment:
```suggestion
	// Remove all the new configurations from the map so that we don't remove one that hasn't
```

Or change it to mark them false, in which case `for upstream := range oldUpstreamsToRemove` needs to check for falses.